### PR TITLE
HW21: implemented JDBC for ItemDao

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: src/main/java/mate/academy/internet/shop/controller/AddItemToBucketController.java [hunk 1]
```diff
@@ -25,7 +25,8 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         Long userId = (Long) req.getSession().getAttribute("userId");
         Bucket bucket = bucketService.getByUserId(userId);
         String itemId = req.getParameter("item_Id");
-        Item item = itemService.get(Long.valueOf(itemId));
+        Item item = null;
+        item = itemService.get(Long.valueOf(itemId));
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controller/AddItemToBucketController.java]: what's the reason of this split?

File: src/main/java/mate/academy/internet/shop/controller/IndexController.java [hunk 2]
```diff
@@ -21,6 +26,7 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        itemService.getAll();
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controller/IndexController.java]: getAll should not be here :) please, remove all test code

File: src/main/java/mate/academy/internet/shop/controller/RemoveItemFromBucketController.java [hunk 3]
```diff
@@ -25,7 +25,8 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         Long userId = (Long) req.getSession().getAttribute("userId");
         Bucket bucket = bucketService.getByUserId(userId);
         String itemId = req.getParameter("item_id");
-        Item item = itemService.get(Long.valueOf(itemId));
+        Item item = null;
+        item = itemService.get(Long.valueOf(itemId));
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/controller/RemoveItemFromBucketController.java]: why? :)

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+import mate.academy.internet.shop.dao.ItemDao;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Item;
+import org.apache.log4j.Logger;
+
+@Dao
+public class ItemDaoJdbcImpl extends AbstractDao<Item> implements ItemDao {
+    private static Logger logger = Logger.getLogger(ItemDaoJdbcImpl.class);
+    private static String DB_Table = "storage.items";
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java]: make it constant, please

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    public Optional<Item> get(Long id) {
+        String query = String.format("SELECT * FROM %s WHERE id = %d;", DB_Table, id);
+        try (Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                Item needed = getItemFromDbTable(resultSet);
+                return Optional.of(needed);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java]: use logger, please

File: src/main/java/mate/academy/internet/shop/controller/AddItemToBucketController.java (hunk 1 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/controller/AddItemToBucketController.java]: sorry, result of unintended and deleted try/catch block

File: src/main/java/mate/academy/internet/shop/controller/IndexController.java (hunk 2 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/controller/IndexController.java]: sorry

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 4 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java]: done. Please, explain why it is important here

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 5 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java]: done

File: src/main/java/mate/academy/internet/shop/controller/RemoveItemFromBucketController.java (hunk 3 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/controller/RemoveItemFromBucketController.java]: sorry, that was Idea`s help result:(((

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 4 above)
Comment (Tarabarnik) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ItemDaoJdbcImpl.java]: @IegorSoboliev There is no good reason to declare a field "public" and "static" without also declaring it "final". Most of the time this is a kludge to share a state among several objects. But with this approach, any object can do whatever it wants with the shared state, such as setting it to null.

**Review by Tarabarnik — APPROVED** [review]

**Pull request merged**